emails/models.py

```diff
@@ -493,14 +493,17 @@ def delete(self, *args, **kwargs):
         return super(RelayAddress, self).delete(*args, **kwargs)
 
     def save(self, *args, **kwargs):
+        profile = self.user.profile_set.first()
         if self._state.adding:
             check_user_can_make_another_address(self.user)
             while True:
                 if valid_address(self.address, self.domain):
                     break
                 self.address = address_default()
-            profile = self.user.profile_set.first()
             profile.update_abuse_metric(address_created=True)
+        if (not profile.server_storage):
+            self.description = ""
```

Comment:
@groovecoder I was wondering if this should also clear `used_on` and, if so, to what value - `None` or `""` or something else?

(Also, is it correct that `generated_for` and `used_on` are not present on DomainAddresses? With the add-on starting to support the latter as well, feels like those fields might need to be present too?)

Reply:
OK, then I took the liberty of also clearing that field when the user disabled `server_storage`. Unfortunately I'm a bit unsure about what will go in that field (I recall seeing a JSON array?), so the mock data in the test may be somewhat idiosyncratic. cb8a5282